script/texdoclib-view.tlu

```diff
@@ -211,31 +216,78 @@ local function print_menu(name, doclist, showall)
     end
 end
 
+local function search_online(name, doclist)
+    if not texdoc.config.get_value('interact_switch') or
+       texdoc.config.get_value('machine_switch')
+    then
+        view_doc(doclist[1])
+        return
+    end
+
+    local function write(msg)
+        io.stderr:write(msg .. '') -- cast to string
+    end
+    -- Fuzzy heuristic to see if the user has _any_ local documentation
+    local texdoc_docs = texdoc.search.get_doclist("texdoc")
+    local docs_installed = texdoc[1] and texdoc[1].basename == "texdoc.pdf"
+
+    if docs_installed then
+        write(C.badmatch_msg:gsub(C.badmatch_msg_ph, name))
+
+        for i=1,3 do
+            local doc = (doclist[i] and doclist[i].normname) or '<empty>'
```

Comment:
We use `realpath` instead of `normname` for showing documents in the regular list (in the `list` mode). Shouldn't we use the same format also for this message?

https://github.com/TeX-Live/texdoc/blob/a27708f201ce4216edf7308d9a92ecdb005b294f/script/texdoclib-view.tlu#L192-L198

Reply:
I think that it makes sense here to use different formats. `texdoc --view` is the "user-friendly" mode, while `texdoc --list` is the comprehensive and detailed mode. With `--list`, the extra path info is useful in case you have multiple TL installations or duplicate files or something, but for most users/usecases, the extra path info is just noise.

This also slightly helps to fit things into one terminal width in the case of really long paths. I think that this:
```text
C:\Users\First Last\texlive\2022\texmf-dist\doc\context\documents\general\manuals\luametatex.pdf
```
would be the longest path that would be plausible, which is 97 characters.

The advantage to `realpath` is that it makes it much clearer what specific document that you are opening, which can be helpful with complicated installations. I don't have a particularly strong opinion here, so just let me know if you'd prefer that I use `realpath` instead of `normname`.

Reply:
Ok. With your explanation, I also think using `normname` is more reasonable than using `realpath`.